plutus-tx-plugin/test/CallTrace/Lib.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications  #-}
+
+module CallTrace.Lib where
+
+import PlutusTx.Code
+
+import Control.Lens ((^.))
+import Data.Text (Text)
+import Prettyprinter (vsep)
+import Test.Tasty (TestName)
+
+import PlutusCore.Evaluation.Machine.ExBudgetingDefaults qualified as PLC
+import PlutusCore.Pretty (Pretty (pretty), Render (render), prettyPlcClassicSimple, prettyReadable)
+import PlutusCore.Test (TestNested, ToUPlc (..), nestedGoldenVsDocM, ppCatch)
+import UntypedPlutusCore qualified as UPLC
+import UntypedPlutusCore.Evaluation.Machine.Cek qualified as UPLC
+import UntypedPlutusCore.Evaluation.Machine.Cek.Internal qualified as UPLC
+
+import PlutusTx.Test (prettyBudget, prettyCodeSize)
+
+goldenEvalCekTraceWithEmitter
+  :: UPLC.EmitterMode UPLC.DefaultUni UPLC.DefaultFun
+  -> TestName
+  -> CompiledCode a
+  -> TestNested
+goldenEvalCekTraceWithEmitter emitter name compiledCode =
+  nestedGoldenVsDocM name ".eval" $ ppCatch $ do
+    uplc <- toUPlc compiledCode
+    let
+      (evalRes, UPLC.CountingSt budget, logOut) =
+        UPLC.runCek
+          PLC.defaultCekParametersForTesting
+          UPLC.counting
+          emitter
+          (uplc ^. UPLC.progTerm)
+
+      traceMsg =
+        case logOut of
+          [] -> ["Trace: <empty>"]
```

Comment:
What if I `trace "<empty>"` ?
Will it be distinguishable from not tracing anything?